Src/Couchbase.Linq/BucketDataStore.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Couchbase.Core;
+using Couchbase.N1QL;
+
+namespace Couchbase.Linq
+{
+    /// <summary>
+    /// A set of lower-level methods for other bucket-related functionality
+    /// </summary>
+    internal class BucketDataStore : IDataStore
+    {
+        private readonly IBucket _bucket;
+
+        internal BucketDataStore(IBucket bucket)
+        {
+            _bucket = bucket;
+        }
+
+        /// <summary>
+        /// Execute a N1QL function. Optionally specify parameters
+        /// </summary>
+        public IQueryResult<T> Execute<T>(string statement, params object[] parameters)
+        {
+            var query = new QueryRequest(statement)
+                .AddPositionalParameter(parameters);
+            return _bucket.Query<T>(query);
```

Comment:
Unfortunately there is no way to modify any of the `QueryRequest` properties (like Timeout or ScanConsistency) if we go this route...at a minimum we should provide an overload which takes a `QueryRequest` or equivalent. Perhaps we need a public wrapper for QueryRequest configuration properties exposed somewhere?